kratos/geometries/hexahedra_3d_27.h

```diff
@@ -1366,6 +1374,331 @@ template<class TPointType> class Hexahedra3D27 : public Geometry<TPointType>
         return d_shape_f_values;
     }
 
+       /**
+     * returns the second order derivatives of all shape functions
+     * in given arbitrary points
+     * @param rResult a third order tensor which contains the second derivatives
+     * @param rPoint the given point the second order derivatives are calculated in
+     */
+    ShapeFunctionsSecondDerivativesType& ShapeFunctionsSecondDerivatives( ShapeFunctionsSecondDerivativesType& rResult, const CoordinatesArrayType& rPoint ) const override
+    {
+        if ( rResult.size() != this->PointsNumber() )
+        {
+            // KLUDGE: While there is a bug in ublas vector resize, I have to put this beside resizing!!
+            ShapeFunctionsGradientsType temp( this->PointsNumber() );
+            rResult.swap( temp );
+        }
+
+        for ( unsigned int i = 0; i < this->PointsNumber(); i++ )
+        {
+            rResult[i].resize( 3, 3, false );
+            noalias( rResult[i] ) = ZeroMatrix( 3, 3 );
```

Comment:
```suggestion
```
As far as I see, below you're assigning the values so there is no need to initialize to zero.

Reply:
I think this one is not committed yet.

Reply:
You are right, I forgot it.